crates/build/src/lib.rs

```diff
@@ -621,16 +644,29 @@ pub fn execute(args: ExecuteArgs) -> Result<BuildResult> {
                 "Building cargo project".bright_green().bold()
             );
             build_steps.increment_current();
-            exec_cargo_for_wasm_target(
+            exec_cargo_for_onchain_target(
                 &crate_metadata,
                 "build",
                 &features,
                 build_mode,
                 network,
                 verbosity,
                 &unstable_flags,
+                target,
             )?;
 
+            // we persist the latest target we used so we trigger a rebuild when we switch
+            fs::write(target_file_path, target.llvm_target())?;
+
+            // we changed the .wasm file ending to the target agnostic .code
+            // in order to not break existing code we create a symlink
+            let symlink = crate_metadata.dest_code.with_extension("wasm");
+            fs::remove_file(&symlink).ok();
+            #[cfg(unix)]
+            if matches!(target, Target::Wasm) {
+                std::os::unix::fs::symlink(&crate_metadata.dest_code, &symlink)?;
+            };
+
```

Comment:
Could we do this on windows too? https://doc.rust-lang.org/stable/std/os/windows/fs/fn.symlink_file.html

Reply:
This documentation says you need admin privileges to create a symlink (lol windows). This is why I didn't do it.

Reply:
No longer relevant. We use arch specific file extensions now which means we keep the wasm file as-is.